### PR TITLE
Change values in instruction metrics

### PR DESCRIPTION
**Pull request by chfast**
This updates instruction metrics with changed values - they are easier to explain and represent number directly usable by VM implementations (will benefit aleth-interepreter).

This is tested against evmone's table: https://github.com/ethereum/evmone/pull/191.

The modification can be easily verified by following the rule:
```
{..., a, b} -> {..., a, b - a}
```

Continuation of https://github.com/ethereum/evmc/issues/356.

### CODE REVIEW

File: include/evmc/instructions.h [hunk 1]
```diff
@@ -188,11 +188,11 @@ struct evmc_instruction_metrics
     /** The instruction gas cost. */
     int16_t gas_cost;
 
-    /** The number of items the instruction pops from the EVM stack before execution. */
-    int8_t num_stack_arguments;
+    /** The number of the EVM stack items required for the instruction. */
```
Comment (gumb0) [inline include/evmc/instructions.h]: "the minimum number" maybe?

File: include/evmc/instructions.h [hunk 2]
```diff
@@ -188,11 +188,11 @@ struct evmc_instruction_metrics
...
+    int8_t stack_height_required;
 
-    /** The number of items the instruction pushes to the EVM stack after execution. */
-    int8_t num_stack_returned_items;
+    /** The EVM stack height change caused by the instruction execution. */
```
Comment (gumb0) [inline include/evmc/instructions.h]: I'd clarify that it's stack after execution - stack before

Comment (chfast) [reply include/evmc/instructions.h]: Added.

File: include/evmc/instructions.h (hunk 1 above)
Comment (chfast) [reply include/evmc/instructions.h]: Changed.

**Pull request merged**